analyzer/restapi-plugin/src/main/java/eu/fasten/analyzer/restapiplugin/server/OpenAPIServer.java

```diff
@@ -0,0 +1,418 @@
+                                            logger.error("Error querying the database: ", e);
+                                            throw e;
+                                        }
+                                    });
+                                });
+
+                // Generate the router
+                Router router = routerFactory.getRouter();
+
+                // Handle errors
+                router.errorHandler(404, routingContext -> {
+                    JsonObject errorObject = new JsonObject()
+                            .put("code", 404)
+                            .put("message",
+                                    (routingContext.failure() != null) ?
+                                            routingContext.failure().getMessage() :
+                                            "Not Found");
+                    routingContext
+                            .response()
+                            .setStatusCode(404)
+                            .putHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+                            .end(errorObject.encode());
+                });
+                router.errorHandler(400, routingContext -> {
+                    JsonObject errorObject = new JsonObject()
+                            .put("code", 400)
+                            .put("message",
+                                    (routingContext.failure() != null) ?
+                                            routingContext.failure().getMessage() :
+                                            "Validation Exception");
+                    routingContext
+                            .response()
+                            .setStatusCode(400)
+                            .putHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+                            .end(errorObject.encode());
+                });
+
+                // Set http server
+                server = vertx.createHttpServer(new HttpServerOptions()
+                        .setPort(8080)
+                        .setHost("localhost"));
+                server.requestHandler(router).listen(serverAsyncResult -> {
+                    if (serverAsyncResult.succeeded()) {
+                        logger.info("Server started on port "
+                                + serverAsyncResult.result().actualPort());
+                    } else {
+                        logger.info(serverAsyncResult.cause());
+                    }
+                });
+                // Complete the verticle start
+                future.complete();
+                logger.info("Verticle start completed");
+            } else {
+                // Something went wrong during router factory initialization
+                future.fail(asyncResult.cause());
+                logger.info("Verticle failed");
+            }
+        });
+    }
+
+    private Timestamp getProperTimestamp(long timestamp) {
```

Comment:
I don't think you actually need this method. It is used in Metadata plugin because some timestamps were in seconds instead of milliseconds and we want to always have milliseconds the database. Since you're only retrieving data from the database, you don't need to adjust any timestamps provided that input timestamp in milliseconds.